src/csync/csync_update.cpp

```diff
@@ -172,14 +171,14 @@ static int _csync_detect_update(CSYNC *ctx, std::unique_ptr<csync_file_stat_t> f
    * does not change on rename.
    */
   if (csync_get_statedb_exists(ctx)) {
-    tmp = csync_statedb_get_stat_by_hash(ctx, fs->phash);
+    tmp = csync_statedb_get_stat_by_path(ctx, fs->path);
 
     if(_last_db_return_error(ctx)) {
         ctx->status_code = CSYNC_STATUS_UNSUCCESSFUL;
         return -1;
     }
 
-    if(tmp && tmp->phash == fs->phash ) { /* there is an entry in the database */
+    if(tmp && tmp->path == fs->path ) { /* there is an entry in the database */
```

Comment:
(minor:  tmp->path == fs->path should actually be an assert)

Reply:
This is changed further by b4b0f63e6d043ceeaec5e9ebf008b94898e4bb04